dash/include/dash/Mutex.h

```diff
@@ -30,50 +31,75 @@ class Mutex {
 private:
   using self_t = Mutex;
 
+  struct DestroyDARTLock {
+    void operator()(dart_lock_t lock)
+    {
+      if (nullptr != lock) {
+        auto ret = dart_team_lock_destroy(&lock);
+
+        if (ret != DART_OK) {
+          DASH_LOG_ERROR(
+              "Failed to destroy DART lock! "
+              "(dart_team_lock_destroy failed)");
+        }
+      }
+    }
+  };
+
 public:
   /**
    * DASH Mutex is only valid for a dash team. If no team is passed, team all
    * is used.
-   * 
+   *
    * This function is not thread-safe
    * @param team team for mutual exclusive accesses
    */
-  explicit Mutex(Team & team = dash::Team::All());
-  
-  Mutex(const Mutex & other)               = delete;
-  Mutex(Mutex && other)                    = default;
-
-  self_t & operator=(const self_t & other) = delete;
-  self_t & operator=(self_t && other)      = default;
-  
+  explicit Mutex(Team& team = dash::Team::All());
+
+  Mutex(const Mutex& other) = delete;
+  Mutex(Mutex&& other)      = default;
+
+  self_t& operator=(const self_t& other) = delete;
+  self_t& operator=(self_t&& other) = default;
+
   /**
    * Collective destructor to destruct a DART lock.
-   * 
+   *
+   * This function is not thread-safe
+   */
+  ~Mutex() = default;
+
+  /**
+   * Collective initialization of the DART lock.
+   *
    * This function is not thread-safe
+   *
+   * @return True if lock was successfully initialized, False otherwise
    */
-  ~Mutex();
-  
+  bool init();
+
   /**
    * Block until the lock was acquired.
    */
   void lock();
-  
+
   /**
    * Try to acquire the lock and return immediately.
    * @return True if lock was successfully aquired, False otherwise
    */
   bool try_lock();
-  
+
   /**
    * Release the lock acquired through \c lock() or \c try_lock().
    */
   void unlock();
-  
-private:
-  dart_lock_t   _mutex;
-}; // class Mutex
 
-} // namespace dash
+private:
+  dash::Team const* _team{nullptr};
+  std::unique_ptr<std::remove_pointer<dart_lock_t>::type, DestroyDARTLock>
+      _mutex{nullptr};
```

Comment:
I would suggest to introduce a constant `DART_LOCK_NULL` (= `(dart_lock_t*)NULL`) and use that instead of `nullptr` and `NULL` in the DASH and DART part as `dart_lock_t` is an opaque type (that happens to be `typedef`ed to a pointer).